src/SqlBindingConfigProvider.cs

```diff
@@ -54,8 +55,27 @@ public void Initialize(ExtensionConfigContext context)
             var converter = new SqlConverter(this._configuration);
             inputOutputRule.BindToInput(converter);
             inputOutputRule.BindToInput<string>(typeof(SqlGenericsConverter<string>), this._configuration, logger);
-            inputOutputRule.BindToCollector<OpenType>(typeof(SqlAsyncCollectorBuilder<>), this._configuration, logger);
+            inputOutputRule.BindToCollector<SQLObjectOpenType>(typeof(SqlAsyncCollectorBuilder<>), this._configuration, logger);
             inputOutputRule.BindToInput<OpenType>(typeof(SqlGenericsConverter<>), this._configuration, logger);
         }
     }
+
+    internal class SQLObjectOpenType : OpenType.Poco
+    {
+        public override bool IsMatch(Type type, OpenTypeMatchContext context)
+        {
+            if (type.IsGenericType
```

Comment:
Please comments explaining what this code is doing

Reply:
To be honest, I don't really know what this code exactly does, there are no comments on the CosmosDB repro either (from where I added this). I added a comment on why I added this, I'll check with them and add further comments.

Reply:
It's best to never add code that we have no idea what it does, so yes please follow up and make sure we have a clear understanding of what's happening here.